lib/drivers/medtronic/processData.js

```diff
@@ -272,24 +272,39 @@ var buildWizardRecords = function (records) {
     bgRaw = bgRaw.concat(struct.extractByte(record.head,1));
     bgInput = struct.extractBEShort(bgRaw,0);
 
-    var carbRaw = [(struct.extractByte(record.body,1) & BITMASKS.RIGHT_1100) >> 2];
+    var carbRaw = [struct.extractByte(record.body,1) >> 2 & 1];
     carbRaw = carbRaw.concat(struct.extractByte(record.body,0));
     carbInput = struct.extractBEShort(carbRaw,0);
 
-    var carbRatioRaw = [struct.extractByte(record.body,2) & BITMASKS.RIGHT_THREE];
+    var carbRatioRaw = [0,struct.extractByte(record.body,9) >> 1 & BITMASKS.RIGHT_TWO];
+    carbRatioRaw = carbRatioRaw.concat(struct.extractByte(record.body,2));
     carbRatioRaw = carbRatioRaw.concat(struct.extractByte(record.body,3));
-    carbRatio = struct.extractBEShort(carbRatioRaw,0) / 10.0;
+    carbRatio = struct.extractBEInt(carbRatioRaw,0) / 10.0;
```

Comment:
We should extract the carb units and divide by 1000.0 instead if it's set to exchanges.